### PR TITLE
Add EditableText scroll into view breaking change note

### PR DESCRIPTION
**Pull request by tgucio**
Breaking change note for https://github.com/flutter/flutter/pull/109114

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/editable-text-scroll-into-view.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+description: >
+  Improve EditableText selection scroll into view behavior to always
+  use the current selection extent.
+---
+
+## Summary
+
+`Editable.onCaretChanged` callback is removed and with this change,
+`EditableText` behavior with regard to scrolling the selection into view
+changes.
```
Comment (sfshaza2) [inline src/release/breaking-changes/editable-text-scroll-into-view.md:12]:
```suggestion
The `Editable.onCaretChanged` callback is removed. With this change,
`EditableText` behavior for scrolling the selection into view
changes.
```

File: src/release/breaking-changes/editable-text-scroll-into-view.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Context
+
+To date, `EditableText` has used multiple mechanisms ensuring the selection
+extent or caret is scrolled into view on updates.
```
Comment (sfshaza2) [inline src/release/breaking-changes/editable-text-scroll-into-view.md:17]: Not sure what you mean by "on updates".

```suggestion
Previously, `EditableText` used multiple mechanisms to determine the
extent of the selection, or to ensure that the caret is scrolled into view.
```

File: src/release/breaking-changes/editable-text-scroll-into-view.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+To date, `EditableText` has used multiple mechanisms ensuring the selection
+extent or caret is scrolled into view on updates. 
+
+## Description of change
+
+By removing the `Editable.onCaretChanged` callback, `EditableText` will always
+use the most up to selection extent location when scrolling to show it.
+Specifically this results in improved scroll into view behavior after
+changing selection from collapsed to non-collapsed via
+`userUpdateTextEditingValue()`.
```
Comment (sfshaza2) [inline src/release/breaking-changes/editable-text-scroll-into-view.md:25]: What does "use the most up to selection extent location" mean?
I've made some tweaks, but would do more once I understand this phrase.

```suggestion
By removing the `Editable.onCaretChanged` callback, `EditableText` will always
use the most up to selection extent location when scrolling to show it.
Specifically, this improves scroll into view behavior after
changing selection from collapsed to non-collapsed using
`userUpdateTextEditingValue()`.
```

File: src/release/breaking-changes/editable-text-scroll-into-view.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## Timeline
+
+Landed in version: 3.12.0-4.0.pre<br>
+In stable release: 3.13.0
```
Comment (sfshaza2) [inline src/release/breaking-changes/editable-text-scroll-into-view.md:30]: The 3.13 release? It's probably best to just say "TBD" for now.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
A few tweaks and a question.

File: src/release/breaking-changes/editable-text-scroll-into-view.md (hunk 2 above)
Comment (tgucio) [reply src/release/breaking-changes/editable-text-scroll-into-view.md:17]: Thanks for the edits. Yes, this is a bit convoluted: what I meant were the mechanisms used on scrolling into view (which happens on user updates) either of:
- selection extent (non-collapsed selection)
- caret position (collapsed selection).

Let me reword as:

> Previously, upon scrolling into view to show user updates, `EditableText` used multiple mechanisms to determine the
> extent of the selection or the caret location.

File: src/release/breaking-changes/editable-text-scroll-into-view.md (hunk 3 above)
Comment (tgucio) [reply src/release/breaking-changes/editable-text-scroll-into-view.md:26]: The underlying issue had to do with using multiple sources of truth where one wasn't necessarily up to date (could be one frame behind).

What is clearly missing is "date" as in "up to date" :). Correcting this.

**Review by atsansone — APPROVED** [review]

**Comment by tgucio** [comment]
@sfshaza2 wondering if you could take another look so we can merge framework's PR 109114? Thanks!

**Review by sfshaza2 — APPROVED** [review]
lgtm.  Thanks, @tgucio!

**Pull request merged**